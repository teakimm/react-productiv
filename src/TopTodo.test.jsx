import { describe, it, expect, test } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const testTodos = [{
    id: "anyId",
    title: "testTitle",
    description: "hello world",
    priority: 3
},
{
    id: "test2",
    title: "this is another test",
    description: "don't read this",
    priority: 1
}
];

it("renders without crashing", function () {
    render(<TopTodo todos={testTodos} />);
});

it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={testTodos} />);
    expect(container).toMatchSnapshot();
});


it("renders correct todo", function () {
    const result = render(<TopTodo todos={testTodos} />);

    expect(
        result.queryByText("this is another test")
    ).toBeInTheDocument();

    expect(
        result.queryByText("hello world")
    ).not.toBeInTheDocument();
});