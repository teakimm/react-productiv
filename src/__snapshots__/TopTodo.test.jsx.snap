// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        this is another test
      </b>
       
      <small>
        (
        priority: 1
        )
      </small>
    </div>
    <div>
      <small>
        don't read this
      </small>
    </div>
  </div>
</div>
`;
